app/mci/pkg/gcp/loadbalancer/loadbalancersyncer_test.go

```diff
@@ -80,25 +109,67 @@ func TestCreateLoadBalancer(t *testing.T) {
 		}
 		return true, ret, nil
 	})
+	// Return a fake ingress with the instance groups annotation.
+	// Return a fake ingress with the instance groups annotation.
+	client.AddReactor("get", "ingresses", func(action core.Action) (handled bool, ret runtime.Object, err error) {
+		return true, &ing, nil
+	})
+
+	// Setup the fake instance group provider to return the expected instance group with fake named ports.
+	err = lbc.igp.CreateInstanceGroup(&compute.InstanceGroup{
+		Name: igName,
+		NamedPorts: []*compute.NamedPort{
+			{
+				Name: portName,
+				Port: nodePort,
+			},
+		},
+	}, igZone)
 	if err := lbc.CreateLoadBalancer(&ing); err != nil {
 		t.Fatalf("unexpected error %s", err)
 	}
+	// Verify client actions.
 	actions := client.Actions()
-	// Verify that the client should have been used to fetch service "my-svc".
-	if len(actions) != 1 {
+	// Verify that the client should have been used to fetch service and ingress.
+	if len(actions) != 2 {
 		t.Fatalf("unexpected actions: %v, expected 1 get", actions)
 	}
-	getName := actions[0].(core.GetAction).GetName()
-	if getName != "my-svc" {
-		t.Fatalf("unexpected get for %s, expected: my-svc", getName)
+	getSvcName := actions[0].(core.GetAction).GetName()
+	if getSvcName != "my-svc" {
+		t.Fatalf("unexpected get for %s, expected: my-svc", getSvcName)
+	}
+	getIngName := actions[1].(core.GetAction).GetName()
+	if getIngName != ing.Name {
+		t.Fatalf("unexpected get for %s, expected: %s", getIngName, ing.Name)
 	}
+	// Verify that the expected healthcheck was created.
 	fhc := lbc.hcs.(*healthcheck.FakeHealthCheckSyncer)
-	// Verify that the expected healthcheck was ensured.
 	if len(fhc.EnsuredHealthChecks) != 1 {
 		t.Fatalf("unexpected number of health checks. expected: %d, got: %d", 1, len(fhc.EnsuredHealthChecks))
 	}
 	hc := fhc.EnsuredHealthChecks[0]
-	if hc.LBName != lbName || hc.Port.Port != int64(nodePort) {
+	if hc.LBName != lbName || hc.Port.Port != nodePort {
 		t.Fatalf("unexpected health check: %v\nexpected: lbname: %s, port: %d", hc, lbName, nodePort)
 	}
+	// Verify that the expected backend service was created.
+	fbs := lbc.bss.(*backendservice.FakeBackendServiceSyncer)
+	if len(fbs.EnsuredBackendServices) != 1 {
+		t.Fatalf("unexpected number of backend services. expected: %d, got: %d", 1, len(fbs.EnsuredBackendServices))
+	}
+	bs := fbs.EnsuredBackendServices[0]
+	if bs.LBName != lbName {
+		t.Fatalf("unexpected lb name in backend service. expected: %s, got: %s", lbName, bs.LBName)
```

Comment:
These fatals can probably also be Errorfs

Reply:
done